### ISSUE TITLE
Field declarations preceded with a label are not parsed properly

### ISSUE THREAD

**Issue by AlekseyTs** [dotnet/roslyn#13712]
```
        [Fact]
        public void AcrossScriptDeclarations()
        {
            string source =
@"int P { get; } = G(""P"");
L:
int F = G(""F"");
System.Console.WriteLine(F);
int Q { get; } = G(""Q"");
static int x = 2;
static int G(string s)
{
    System.Console.WriteLine(""{0}: {1}"", x, s);
    x++;
    return x;
}
if (Q < 4) goto L;";
            string expectedOutput =
@"2: P
3: F
4: Q";
            var compilation = CreateCompilationWithMscorlib45(source, references: new[] { SystemCoreRef }, parseOptions: TestOptions.Script, options: TestOptions.DebugExe);
            compilation.VerifyDiagnostics();
            CompileAndVerify(compilation, expectedOutput: expectedOutput, verify: false);
        }
```

Observed:
```
                // (4,26): error CS0103: The name 'F' does not exist in the current context
                // System.Console.WriteLine(F);
                Diagnostic(ErrorCode.ERR_NameNotInContext, "F").WithArguments("F").WithLocation(4, 26)
```

Expected: No errors.

**Comment by AlekseyTs** [comment]
CC @tmat

**Comment by AlekseyTs** [comment]
The issue is even worse, if ```System.Console.WriteLine(F);``` line is removed and TestOptions.ReleaseExe is used, compiler crashes:
```
Test 'Tests.AcrossScriptDeclarations' failed: System.Collections.Generic.KeyNotFoundException : The given key was not present in the dictionary.
	at System.Collections.Generic.Dictionary`2.get_Item(TKey key)
	CodeGen\Optimizer.cs(1761,0): at Microsoft.CodeAnalysis.CSharp.CodeGen.StackOptimizerPass1.RecordVarWrite(LocalSymbol local)
	CodeGen\Optimizer.cs(958,0): at Microsoft.CodeAnalysis.CSharp.CodeGen.StackOptimizerPass1.VisitAssignmentOperator(BoundAssignmentOperator node)
	F:\GitHub\AlekseyTs\roslyn\Binaries\Obj\CSharpCodeAnalysis\Debug\BoundNodes.xml.Generated.cs(1148,0): at Microsoft.CodeAnalysis.CSharp.BoundAssignmentOperator.Accept(BoundTreeVisitor visitor)
	BoundTree\BoundTreeVisitors.cs(147,0): at Microsoft.CodeAnalysis.CSharp.BoundTreeVisitor.Visit(BoundNode node)
	CodeGen\Optimizer.cs(464,0): at Microsoft.CodeAnalysis.CSharp.CodeGen.StackOptimizerPass1.VisitExpressionCore(BoundExpression node, ExprContext context)
	CodeGen\Optimizer.cs(520,0): at Microsoft.CodeAnalysis.CSharp.CodeGen.StackOptimizerPass1.VisitExpressionCoreWithStackGuard(BoundExpression node, ExprContext context)
	CodeGen\Optimizer.cs(507,0): at Microsoft.CodeAnalysis.CSharp.CodeGen.StackOptimizerPass1.VisitExpression(BoundExpression node, ExprContext context)
	CodeGen\Optimizer.cs(817,0): at Microsoft.CodeAnalysis.CSharp.CodeGen.StackOptimizerPass1.VisitExpressionStatement(BoundExpressionStatement node)
	F:\GitHub\AlekseyTs\roslyn\Binaries\Obj\CSharpCodeAnalysis\Debug\BoundNodes.xml.Generated.cs(2742,0): at Microsoft.CodeAnalysis.CSharp.BoundExpressionStatement.Accept(BoundTreeVisitor visitor)
	BoundTree\BoundTreeVisitors.cs(147,0): at Microsoft.CodeAnalysis.CSharp.BoundTreeVisitor.Visit(BoundNode node)
	CodeGen\Optimizer.cs(573,0): at Microsoft.CodeAnalysis.CSharp.CodeGen.StackOptimizerPass1.VisitStatement(BoundNode node)
	CodeGen\Optimizer.cs(451,0): at Microsoft.CodeAnalysis.CSharp.CodeGen.StackOptimizerPass1.Visit(BoundNode node)
	F:\GitHub\AlekseyTs\roslyn\Binaries\Obj\CSharpCodeAnalysis\Debug\BoundNodes.xml.Generated.cs(8778,0): at Microsoft.CodeAnalysis.CSharp.BoundTreeRewriter.VisitSequencePointWithSpan(BoundSequencePointWithSpan node)
	F:\GitHub\AlekseyTs\roslyn\Binaries\Obj\CSharpCodeAnalysis\Debug\BoundNodes.xml.Generated.cs(2351,0): at Microsoft.CodeAnalysis.CSharp.BoundSequencePointWithSpan.Accept(BoundTreeVisitor visitor)
	BoundTree\BoundTreeVisitors.cs(147,0): at Microsoft.CodeAnalysis.CSharp.BoundTreeVisitor.Visit(BoundNode node)
	CodeGen\Optimizer.cs(573,0): at Microsoft.CodeAnalysis.CSharp.CodeGen.StackOptimizerPass1.VisitStatement(BoundNode node)
	CodeGen\Optimizer.cs(451,0): at Microsoft.CodeAnalysis.CSharp.CodeGen.StackOptimizerPass1.Visit(BoundNode node)
	BoundTree\BoundTreeRewriter.cs(36,0): at Microsoft.CodeAnalysis.CSharp.BoundTreeRewriter.DoVisitList[T](ImmutableArray`1 list)
	BoundTree\BoundTreeRewriter.cs(25,0): at Microsoft.CodeAnalysis.CSharp.BoundTreeRewriter.VisitList[T](ImmutableArray`1 list)
	F:\GitHub\AlekseyTs\roslyn\Binaries\Obj\CSharpCodeAnalysis\Debug\BoundNodes.xml.Generated.cs(9031,0): at Microsoft.CodeAnalysis.CSharp.BoundTreeRewriter.VisitStatementList(BoundStatementList node)
	F:\GitHub\AlekseyTs\roslyn\Binaries\Obj\CSharpCodeAnalysis\Debug\BoundNodes.xml.Generated.cs(4096,0): at Microsoft.CodeAnalysis.CSharp.BoundStatementList.Accept(BoundTreeVisitor visitor)
	BoundTree\BoundTreeVisitors.cs(147,0): at Microsoft.CodeAnalysis.CSharp.BoundTreeVisitor.Visit(BoundNode node)
	CodeGen\Optimizer.cs(573,0): at Microsoft.CodeAnalysis.CSharp.CodeGen.StackOptimizerPass1.VisitStatement(BoundNode node)
	CodeGen\Optimizer.cs(451,0): at Microsoft.CodeAnalysis.CSharp.CodeGen.StackOptimizerPass1.Visit(BoundNode node)
	BoundTree\BoundTreeRewriter.cs(36,0): at Microsoft.CodeAnalysis.CSharp.BoundTreeRewriter.DoVisitList[T](ImmutableArray`1 list)
	BoundTree\BoundTreeRewriter.cs(25,0): at Microsoft.CodeAnalysis.CSharp.BoundTreeRewriter.VisitList[T](ImmutableArray`1 list)
	F:\GitHub\AlekseyTs\roslyn\Binaries\Obj\CSharpCodeAnalysis\Debug\BoundNodes.xml.Generated.cs(8783,0): at Microsoft.CodeAnalysis.CSharp.BoundTreeRewriter.VisitBlock(BoundBlock node)
	CodeGen\Optimizer.cs(597,0): at Microsoft.CodeAnalysis.CSharp.CodeGen.StackOptimizerPass1.VisitBlock(BoundBlock node)
	F:\GitHub\AlekseyTs\roslyn\Binaries\Obj\CSharpCodeAnalysis\Debug\BoundNodes.xml.Generated.cs(2387,0): at Microsoft.CodeAnalysis.CSharp.BoundBlock.Accept(BoundTreeVisitor visitor)
	BoundTree\BoundTreeVisitors.cs(147,0): at Microsoft.CodeAnalysis.CSharp.BoundTreeVisitor.Visit(BoundNode node)
	CodeGen\Optimizer.cs(573,0): at Microsoft.CodeAnalysis.CSharp.CodeGen.StackOptimizerPass1.VisitStatement(BoundNode node)
	CodeGen\Optimizer.cs(451,0): at Microsoft.CodeAnalysis.CSharp.CodeGen.StackOptimizerPass1.Visit(BoundNode node)
	BoundTree\BoundTreeRewriter.cs(36,0): at Microsoft.CodeAnalysis.CSharp.BoundTreeRewriter.DoVisitList[T](ImmutableArray`1 list)
	BoundTree\BoundTreeRewriter.cs(25,0): at Microsoft.CodeAnalysis.CSharp.BoundTreeRewriter.VisitList[T](ImmutableArray`1 list)
	F:\GitHub\AlekseyTs\roslyn\Binaries\Obj\CSharpCodeAnalysis\Debug\BoundNodes.xml.Generated.cs(8783,0): at Microsoft.CodeAnalysis.CSharp.BoundTreeRewriter.VisitBlock(BoundBlock node)
	CodeGen\Optimizer.cs(597,0): at Microsoft.CodeAnalysis.CSharp.CodeGen.StackOptimizerPass1.VisitBlock(BoundBlock node)
	F:\GitHub\AlekseyTs\roslyn\Binaries\Obj\CSharpCodeAnalysis\Debug\BoundNodes.xml.Generated.cs(2387,0): at Microsoft.CodeAnalysis.CSharp.BoundBlock.Accept(BoundTreeVisitor visitor)
	BoundTree\BoundTreeVisitors.cs(147,0): at Microsoft.CodeAnalysis.CSharp.BoundTreeVisitor.Visit(BoundNode node)
	CodeGen\Optimizer.cs(573,0): at Microsoft.CodeAnalysis.CSharp.CodeGen.StackOptimizerPass1.VisitStatement(BoundNode node)
	CodeGen\Optimizer.cs(451,0): at Microsoft.CodeAnalysis.CSharp.CodeGen.StackOptimizerPass1.Visit(BoundNode node)
	CodeGen\Optimizer.cs(1481,0): at Microsoft.CodeAnalysis.CSharp.CodeGen.StackOptimizerPass1.VisitTryStatement(BoundTryStatement node)
	F:\GitHub\AlekseyTs\roslyn\Binaries\Obj\CSharpCodeAnalysis\Debug\BoundNodes.xml.Generated.cs(3513,0): at Microsoft.CodeAnalysis.CSharp.BoundTryStatement.Accept(BoundTreeVisitor visitor)
	BoundTree\BoundTreeVisitors.cs(147,0): at Microsoft.CodeAnalysis.CSharp.BoundTreeVisitor.Visit(BoundNode node)
	CodeGen\Optimizer.cs(573,0): at Microsoft.CodeAnalysis.CSharp.CodeGen.StackOptimizerPass1.VisitStatement(BoundNode node)
	CodeGen\Optimizer.cs(451,0): at Microsoft.CodeAnalysis.CSharp.CodeGen.StackOptimizerPass1.Visit(BoundNode node)
	BoundTree\BoundTreeRewriter.cs(36,0): at Microsoft.CodeAnalysis.CSharp.BoundTreeRewriter.DoVisitList[T](ImmutableArray`1 list)
	BoundTree\BoundTreeRewriter.cs(25,0): at Microsoft.CodeAnalysis.CSharp.BoundTreeRewriter.VisitList[T](ImmutableArray`1 list)
	F:\GitHub\AlekseyTs\roslyn\Binaries\Obj\CSharpCodeAnalysis\Debug\BoundNodes.xml.Generated.cs(8783,0): at Microsoft.CodeAnalysis.CSharp.BoundTreeRewriter.VisitBlock(BoundBlock node)
	CodeGen\Optimizer.cs(597,0): at Microsoft.CodeAnalysis.CSharp.CodeGen.StackOptimizerPass1.VisitBlock(BoundBlock node)
	F:\GitHub\AlekseyTs\roslyn\Binaries\Obj\CSharpCodeAnalysis\Debug\BoundNodes.xml.Generated.cs(2387,0): at Microsoft.CodeAnalysis.CSharp.BoundBlock.Accept(BoundTreeVisitor visitor)
	BoundTree\BoundTreeVisitors.cs(147,0): at Microsoft.CodeAnalysis.CSharp.BoundTreeVisitor.Visit(BoundNode node)
	CodeGen\Optimizer.cs(573,0): at Microsoft.CodeAnalysis.CSharp.CodeGen.StackOptimizerPass1.VisitStatement(BoundNode node)
	CodeGen\Optimizer.cs(451,0): at Microsoft.CodeAnalysis.CSharp.CodeGen.StackOptimizerPass1.Visit(BoundNode node)
	F:\GitHub\AlekseyTs\roslyn\Binaries\Obj\CSharpCodeAnalysis\Debug\BoundNodes.xml.Generated.cs(8767,0): at Microsoft.CodeAnalysis.CSharp.BoundTreeRewriter.VisitSequencePoint(BoundSequencePoint node)
	F:\GitHub\AlekseyTs\roslyn\Binaries\Obj\CSharpCodeAnalysis\Debug\BoundNodes.xml.Generated.cs(2289,0): at Microsoft.CodeAnalysis.CSharp.BoundSequencePoint.Accept(BoundTreeVisitor visitor)
	BoundTree\BoundTreeVisitors.cs(147,0): at Microsoft.CodeAnalysis.CSharp.BoundTreeVisitor.Visit(BoundNode node)
	CodeGen\Optimizer.cs(573,0): at Microsoft.CodeAnalysis.CSharp.CodeGen.StackOptimizerPass1.VisitStatement(BoundNode node)
	CodeGen\Optimizer.cs(451,0): at Microsoft.CodeAnalysis.CSharp.CodeGen.StackOptimizerPass1.Visit(BoundNode node)
	BoundTree\BoundTreeRewriter.cs(36,0): at Microsoft.CodeAnalysis.CSharp.BoundTreeRewriter.DoVisitList[T](ImmutableArray`1 list)
	BoundTree\BoundTreeRewriter.cs(25,0): at Microsoft.CodeAnalysis.CSharp.BoundTreeRewriter.VisitList[T](ImmutableArray`1 list)
	F:\GitHub\AlekseyTs\roslyn\Binaries\Obj\CSharpCodeAnalysis\Debug\BoundNodes.xml.Generated.cs(8783,0): at Microsoft.CodeAnalysis.CSharp.BoundTreeRewriter.VisitBlock(BoundBlock node)
	CodeGen\Optimizer.cs(597,0): at Microsoft.CodeAnalysis.CSharp.CodeGen.StackOptimizerPass1.VisitBlock(BoundBlock node)
	F:\GitHub\AlekseyTs\roslyn\Binaries\Obj\CSharpCodeAnalysis\Debug\BoundNodes.xml.Generated.cs(2387,0): at Microsoft.CodeAnalysis.CSharp.BoundBlock.Accept(BoundTreeVisitor visitor)
	BoundTree\BoundTreeVisitors.cs(147,0): at Microsoft.CodeAnalysis.CSharp.BoundTreeVisitor.Visit(BoundNode node)
	CodeGen\Optimizer.cs(573,0): at Microsoft.CodeAnalysis.CSharp.CodeGen.StackOptimizerPass1.VisitStatement(BoundNode node)
	CodeGen\Optimizer.cs(451,0): at Microsoft.CodeAnalysis.CSharp.CodeGen.StackOptimizerPass1.Visit(BoundNode node)
	CodeGen\Optimizer.cs(433,0): at Microsoft.CodeAnalysis.CSharp.CodeGen.StackOptimizerPass1.Analyze(BoundNode node, Dictionary`2 locals, Boolean debugFriendly)
	CodeGen\Optimizer.cs(60,0): at Microsoft.CodeAnalysis.CSharp.CodeGen.Optimizer.Optimize(BoundStatement src, Boolean debugFriendly, HashSet`1& stackLocals)
	CodeGen\CodeGenerator.cs(115,0): at Microsoft.CodeAnalysis.CSharp.CodeGen.CodeGenerator..ctor(MethodSymbol method, BoundStatement boundBody, ILBuilder builder, PEModuleBuilder moduleBuilder, DiagnosticBag diagnostics, OptimizationLevel optimizations, Boolean emittingPdb)
	Compiler\MethodCompiler.cs(1332,0): at Microsoft.CodeAnalysis.CSharp.MethodCompiler.GenerateMethodBody(PEModuleBuilder moduleBuilder, MethodSymbol method, Int32 methodOrdinal, BoundStatement block, ImmutableArray`1 lambdaDebugInfo, ImmutableArray`1 closureDebugInfo, StateMachineTypeSymbol stateMachineTypeOpt, VariableSlotAllocator variableSlotAllocatorOpt, DiagnosticBag diagnostics, DebugDocumentProvider debugDocumentProvider, ImportChain importChainOpt, Boolean emittingPdb, ImmutableArray`1 dynamicAnalysisSpans)
	Compiler\MethodCompiler.cs(631,0): at Microsoft.CodeAnalysis.CSharp.MethodCompiler.CompileSynthesizedMethods(TypeCompilationState compilationState)
	Compiler\MethodCompiler.cs(537,0): at Microsoft.CodeAnalysis.CSharp.MethodCompiler.CompileNamedType(NamedTypeSymbol containingType)
	Compiler\MethodCompiler.cs(323,0): at Microsoft.CodeAnalysis.CSharp.MethodCompiler.<>c__DisplayClass20_0.<CompileNamedTypeAsTask>b__0()
	InternalUtilities\UICultureUtilities.cs(136,0): at Roslyn.Utilities.UICultureUtilities.<>c__DisplayClass5_0.<WithCurrentUICulture>b__0()
	at System.Threading.Tasks.Task.InnerInvoke()
	at System.Threading.Tasks.Task.Execute()
	--- End of stack trace from previous location where exception was thrown ---
	at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
	at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
	at System.Runtime.CompilerServices.TaskAwaiter.GetResult()
	Compiler\MethodCompiler.cs(238,0): at Microsoft.CodeAnalysis.CSharp.MethodCompiler.WaitForWorkers()
	Compiler\MethodCompiler.cs(137,0): at Microsoft.CodeAnalysis.CSharp.MethodCompiler.CompileMethodBodies(CSharpCompilation compilation, PEModuleBuilder moduleBeingBuiltOpt, Boolean generateDebugInfo, Boolean hasDeclarationErrors, DiagnosticBag diagnostics, Predicate`1 filterOpt, CancellationToken cancellationToken)
	Compilation\CSharpCompilation.cs(2335,0): at Microsoft.CodeAnalysis.CSharp.CSharpCompilation.CompileMethods(CommonPEModuleBuilder moduleBuilder, Boolean emittingPdb, DiagnosticBag diagnostics, Predicate`1 filterOpt, CancellationToken cancellationToken)
	Compilation\Compilation.cs(1905,0): at Microsoft.CodeAnalysis.Compilation.Emit(Stream peStream, Stream pdbStream, Stream xmlDocumentationStream, Stream win32Resources, IEnumerable`1 manifestResources, EmitOptions options, IMethodSymbol debugEntryPoint, Stream sourceLinkStream, IEnumerable`1 embeddedTexts, CompilationTestData testData, CancellationToken cancellationToken)
	F:\GitHub\AlekseyTs\roslyn\src\Test\Utilities\Shared\Compilation\IRuntimeEnvironment.cs(211,0): at Roslyn.Test.Utilities.RuntimeUtilities.EmitCompilationCore(Compilation compilation, IEnumerable`1 manifestResources, DiagnosticBag diagnostics, CompilationTestData testData, EmitOptions emitOptions)
	F:\GitHub\AlekseyTs\roslyn\src\Test\Utilities\Shared\Compilation\IRuntimeEnvironment.cs(189,0): at Roslyn.Test.Utilities.RuntimeUtilities.EmitCompilation(Compilation compilation, IEnumerable`1 manifestResources, List`1 dependencies, DiagnosticBag diagnostics, CompilationTestData testData, EmitOptions emitOptions)
	HostedRuntimeEnvironment.cs(199,0): at Microsoft.CodeAnalysis.Test.Utilities.HostedRuntimeEnvironment.Emit(Compilation mainCompilation, IEnumerable`1 manifestResources, EmitOptions emitOptions, Boolean usePdbForDebugging)
	CommonTestBase.CompilationVerifier.cs(128,0): at Microsoft.CodeAnalysis.Test.Utilities.CommonTestBase.CompilationVerifier.Emit(IRuntimeEnvironment testEnvironment, IEnumerable`1 manifestResources, EmitOptions emitOptions)
	CommonTestBase.CompilationVerifier.cs(94,0): at Microsoft.CodeAnalysis.Test.Utilities.CommonTestBase.CompilationVerifier.Emit(String expectedOutput, IEnumerable`1 manifestResources, EmitOptions emitOptions, Boolean peVerify, SignatureDescription[] expectedSignatures)
	F:\GitHub\AlekseyTs\roslyn\src\Test\Utilities\Desktop\CommonTestBase.cs(212,0): at Microsoft.CodeAnalysis.Test.Utilities.CommonTestBase.Emit(Compilation compilation, IEnumerable`1 dependencies, IEnumerable`1 manifestResources, SignatureDescription[] expectedSignatures, String expectedOutput, Action`1 assemblyValidator, Action`1 symbolValidator, EmitOptions emitOptions, Boolean verify)
	F:\GitHub\AlekseyTs\roslyn\src\Test\Utilities\Desktop\CommonTestBase.cs(135,0): at Microsoft.CodeAnalysis.Test.Utilities.CommonTestBase.CompileAndVerify(Compilation compilation, IEnumerable`1 manifestResources, IEnumerable`1 dependencies, Action`1 sourceSymbolValidator, Action`1 assemblyValidator, Action`1 symbolValidator, SignatureDescription[] expectedSignatures, String expectedOutput, EmitOptions emitOptions, Boolean verify)
	F:\GitHub\AlekseyTs\roslyn\src\Compilers\Test\Utilities\CSharp.Desktop\CSharpTestBase.cs(200,0): at Microsoft.CodeAnalysis.CSharp.Test.Utilities.CSharpTestBase.CompileAndVerify(Compilation compilation, IEnumerable`1 manifestResources, IEnumerable`1 dependencies, Action`1 sourceSymbolValidator, Action`1 validator, Action`1 symbolValidator, SignatureDescription[] expectedSignatures, String expectedOutput, Boolean verify)
Tests.cs(24048,0): at Tests.AcrossScriptDeclarations()
```
